### ISSUE TITLE
Prettier should print trailing commas after type parameters

### ISSUE THREAD

**Issue by so1ve** [prettier/prettier#15264]
**Prettier 3.0.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAYhCbAXmwCEBDAJwB4AVTOARgBpt6sAmAPgG4AdKADMC1fAVYVKlLgAoAlPyEBXKGBgBLaNmERRBCBKrT52YAF8B62HEqDyYHJL3iyRrqbOL0WVwC9nBq5S7iTmigD04dgAklDYkLCU5ADOMKyCKmqaUMkAdPkCOjI6LgBGRpQKhZkaWkU6rOWVHiDMIBAYtTnIoEYQAO4AClQIycgg5AA2-eRoY22lSWAA1nAwAMrkALZwADJWcMh2k8lwC0urGxj2VgDmyDCUSmcgp1vqD08vyXeTcACKSgg8COU1ObQAVskAB7rX4AoEgpDHcEgACOiLgg0oHTGKBSAFooHA4AATMmtVBJdSTO4AYQgWy25HGU0mlJ+UFufwAgjBHupSkp4IMbPtiaCTi8ABYwLaTADq0vU8GS1wc61GKvUADcVWhxmBkvMQDrnrFybB1mBKOpOjyoKT1ug-pLURgcacFUkMOMPXBTpQdYc2lZAzBseRbsy3S9rpRA+NvAGbXaYJSPVYYAr1KSYNLkIwAAxtShwDHqMuR6Ms5Fgl4wcilHN5gtIDhtJSnWhNvEol5wLalMnk0m7chcpRRuD4SjM-l3VnCiAgMxmIA)

```sh
--parser typescript
--print-width 10
```

**Input:**

```tsx
type Foo = Bar<Type1, Type2>;
foo<Fooo, Barr>();
function foo<Fooo, Barr>() {}
interface Bar<Fooo, Barr> {};
type Baz<Fooo, Barr> = {};
// In contrast, functions...
foo(foooo, barrr);
function foo(foo, bar) {}
```

**Output:**

```tsx
type Foo =
  Bar<
    Type1,
    Types // Here
  >;
foo<
  Fooo,
  Barr // Why?
>();
function foo<
  Fooo,
  Barr,
>() {}
interface Bar<
  Fooo,
  Barr,
> {}
type Baz<
  Fooo,
  Barr,
> = {};
// In contrast, functions...
foo(
  foooo,
  barrr, // Arguments...
);
function foo(
  foo,
  bar, // Parameters...
) {}

```

**Expected behavior:**

Prettier should print trailing commas after type parameters.

**Comment by santosh-kuchetti** [comment]
@sosukesuzuki you mean our expected behaviour is

**Comment by santosh-kuchetti** [comment]
@sosukesuzuki  maybe you are getting that since you provided the print width as 10. if we provide the width lill more it work fine

**Comment by so1ve** [comment]
Print width is irrelevant. I just want to make these generics break into multiple lines.

BTW you are @ the wrong personally

**Comment by sosukesuzuki** [comment]
Yes, a smaller print-width is useful for easily reproducing Prettier's line breaking and trailing comma bugs.

**Comment by so1ve** [comment]
Sorry, after more investigation I found adding trailing commas here triggers `Trailing comma not allowed. ts(1009)`. Closing
